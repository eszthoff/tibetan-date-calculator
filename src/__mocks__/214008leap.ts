export default `{
    "year": 2140,
    "month": 8,
    "isLeapMonth": true,
    "isDoubledMonth": true,
    "startDate": "2013-09-06",
    "endDate": "2013-10-04",
    "days": {
        "2140-8-leap-1": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 1,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-06"
        },
        "2140-8-leap-2": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 2,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-07"
        },
        "2140-8-leap-3": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 3,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-08"
        },
        "2140-8-leap-4": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 4,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-09"
        },
        "2140-8-leap-5": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 5,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-10"
        },
        "2140-8-leap-6": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 6,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-11"
        },
        "2140-8-leap-7": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 7,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-12"
        },
        "2140-8-leap-8": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 8,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-13"
        },
        "2140-8-leap-9": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 9,
            "skippedDay": true,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-13"
        },
        "2140-8-leap-10": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 10,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-14"
        },
        "2140-8-leap-11": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 11,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-15"
        },
        "2140-8-leap-12": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 12,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-16"
        },
        "2140-8-leap-13": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 13,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-17"
        },
        "2140-8-leap-14": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 14,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-18"
        },
        "2140-8-leap-15": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 15,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-19"
        },
        "2140-8-leap-16": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 16,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-20"
        },
        "2140-8-leap-17": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 17,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-21"
        },
        "2140-8-leap-18": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 18,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-22"
        },
        "2140-8-leap-19": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 19,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-23"
        },
        "2140-8-leap-20": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 20,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-24"
        },
        "2140-8-leap-21": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 21,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-25"
        },
        "2140-8-leap-22": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 22,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-26"
        },
        "2140-8-leap-23": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 23,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-27"
        },
        "2140-8-leap-24": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 24,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-28"
        },
        "2140-8-leap-25": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 25,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-29"
        },
        "2140-8-leap-26": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 26,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-09-30"
        },
        "2140-8-leap-27": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 27,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-01"
        },
        "2140-8-leap-28": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 28,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-02"
        },
        "2140-8-leap-29": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 29,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-03"
        },
        "2140-8-leap-30": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": true,
                "isDoubledMonth": true
            },
            "day": 30,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-04"
        }
    },
    "westernIndex": {
        "2013-09-06": "2140-8-leap-1",
        "2013-09-07": "2140-8-leap-2",
        "2013-09-08": "2140-8-leap-3",
        "2013-09-09": "2140-8-leap-4",
        "2013-09-10": "2140-8-leap-5",
        "2013-09-11": "2140-8-leap-6",
        "2013-09-12": "2140-8-leap-7",
        "2013-09-13": "2140-8-leap-8",
        "2013-09-14": "2140-8-leap-10",
        "2013-09-15": "2140-8-leap-11",
        "2013-09-16": "2140-8-leap-12",
        "2013-09-17": "2140-8-leap-13",
        "2013-09-18": "2140-8-leap-14",
        "2013-09-19": "2140-8-leap-15",
        "2013-09-20": "2140-8-leap-16",
        "2013-09-21": "2140-8-leap-17",
        "2013-09-22": "2140-8-leap-18",
        "2013-09-23": "2140-8-leap-19",
        "2013-09-24": "2140-8-leap-20",
        "2013-09-25": "2140-8-leap-21",
        "2013-09-26": "2140-8-leap-22",
        "2013-09-27": "2140-8-leap-23",
        "2013-09-28": "2140-8-leap-24",
        "2013-09-29": "2140-8-leap-25",
        "2013-09-30": "2140-8-leap-26",
        "2013-10-01": "2140-8-leap-27",
        "2013-10-02": "2140-8-leap-28",
        "2013-10-03": "2140-8-leap-29",
        "2013-10-04": "2140-8-leap-30"
    }
}`;
