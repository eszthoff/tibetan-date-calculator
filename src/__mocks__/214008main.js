export default `{
    "year": 2140,
    "month": 8,
    "isLeapMonth": false,
    "isDoubledMonth": true,
    "startDate": "2013-10-05",
    "endDate": "2013-11-03",
    "days": {
        "2140-8-main-1": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 1,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-05"
        },
        "2140-8-main-2": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 2,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-06"
        },
        "2140-8-main-3": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 3,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-07"
        },
        "2140-8-main-4": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 4,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-08"
        },
        "2140-8-main-5": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 5,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-09"
        },
        "2140-8-main-6": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 6,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-10"
        },
        "2140-8-main-7": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 7,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-11"
        },
        "2140-8-main-8": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 8,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-12"
        },
        "2140-8-main-9": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 9,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-13"
        },
        "2140-8-main-10": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 10,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-14"
        },
        "2140-8-main-11": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 11,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-15"
        },
        "2140-8-main-12": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 12,
            "skippedDay": true,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-15"
        },
        "2140-8-main-13": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 13,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-16"
        },
        "2140-8-main-14": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 14,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-17"
        },
        "2140-8-main-15": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 15,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-18"
        },
        "2140-8-main-16": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 16,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-19"
        },
        "2140-8-main-17": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 17,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-20"
        },
        "2140-8-main-18": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 18,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-21"
        },
        "2140-8-main-19": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 19,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-22"
        },
        "2140-8-main-20": {
            "doubled": true
        },
        "2140-8-main-20-main": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 20,
            "skippedDay": false,
            "isLeapDay": true,
            "isDoubledDay": true,
            "westernDate": "2013-10-23"
        },
        "2140-8-main-20-leap": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 20,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": true,
            "westernDate": "2013-10-24"
        },
        "2140-8-main-21": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 21,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-25"
        },
        "2140-8-main-22": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 22,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-26"
        },
        "2140-8-main-23": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 23,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-27"
        },
        "2140-8-main-24": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 24,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-28"
        },
        "2140-8-main-25": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 25,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-29"
        },
        "2140-8-main-26": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 26,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-30"
        },
        "2140-8-main-27": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 27,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-10-31"
        },
        "2140-8-main-28": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 28,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-11-01"
        },
        "2140-8-main-29": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 29,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-11-02"
        },
        "2140-8-main-30": {
            "year": 2140,
            "month": {
                "month": 8,
                "isLeapMonth": false,
                "isDoubledMonth": true
            },
            "day": 30,
            "skippedDay": false,
            "isLeapDay": false,
            "isDoubledDay": false,
            "westernDate": "2013-11-03"
        }
    },
    "westernIndex": {
        "2013-10-05": "2140-8-main-1",
        "2013-10-06": "2140-8-main-2",
        "2013-10-07": "2140-8-main-3",
        "2013-10-08": "2140-8-main-4",
        "2013-10-09": "2140-8-main-5",
        "2013-10-10": "2140-8-main-6",
        "2013-10-11": "2140-8-main-7",
        "2013-10-12": "2140-8-main-8",
        "2013-10-13": "2140-8-main-9",
        "2013-10-14": "2140-8-main-10",
        "2013-10-15": "2140-8-main-11",
        "2013-10-16": "2140-8-main-13",
        "2013-10-17": "2140-8-main-14",
        "2013-10-18": "2140-8-main-15",
        "2013-10-19": "2140-8-main-16",
        "2013-10-20": "2140-8-main-17",
        "2013-10-21": "2140-8-main-18",
        "2013-10-22": "2140-8-main-19",
        "2013-10-23": "2140-8-main-20-main",
        "2013-10-24": "2140-8-main-20-leap",
        "2013-10-25": "2140-8-main-21",
        "2013-10-26": "2140-8-main-22",
        "2013-10-27": "2140-8-main-23",
        "2013-10-28": "2140-8-main-24",
        "2013-10-29": "2140-8-main-25",
        "2013-10-30": "2140-8-main-26",
        "2013-10-31": "2140-8-main-27",
        "2013-11-01": "2140-8-main-28",
        "2013-11-02": "2140-8-main-29",
        "2013-11-03": "2140-8-main-30"
    }
}`;
