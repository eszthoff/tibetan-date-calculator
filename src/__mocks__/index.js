const dayInDuplicateMonthMock = {
  year: 2140,
  month: 8,
  day: 4,
  julianLeap: 2456574,
  julianMain: 2456545,
  westernLeap: new Date('2013-09-09T12:00:00.000Z'),
  westernMain: new Date('2013-10-08T12:00:00.000Z'),
  dayInLeapMonth: {
    day: 4,
    hasLeapDay: false,
    isLeapDay: false,
    julianDate: 2456545,
    month: {
      hasLeapMonth: true, isLeapMonth: true, month: 8, year: 2140
    },
    skippedDay: false,
    westernDate: new Date('2013-09-09T12:00:00.000Z'),
    year: {
      rabjungCycle: 17,
      rabjungYear: 27,
      tibYear: 2140,
      westernYear: 2013,
      animal: 'Snake',
      element: 'Water',
      gender: 'Female',
    }
  },
  dayInMainMonth: {
    day: 4,
    hasLeapDay: false,
    isLeapDay: false,
    julianDate: 2456574,
    month: {
      hasLeapMonth: true, isLeapMonth: false, month: 8, year: 2140
    },
    skippedDay: false,
    westernDate: new Date('2013-10-08T12:00:00.000Z'),
    year: {
      rabjungCycle: 17,
      rabjungYear: 27,
      tibYear: 2140,
      westernYear: 2013,
      animal: 'Snake',
      element: 'Water',
      gender: 'Female',
    }
  },
};

const skippedDayMock = {
  year: 2140,
  month: 2,
  day: 20,
  western: new Date('2013-03-31T12:00:00.000Z'),
  dayObject: {
    day: 20,
    hasLeapDay: false,
    isLeapDay: false,
    julianDate: 2456383,
    month: {
      hasLeapMonth: false, isLeapMonth: false, month: 2, year: 2140
    },
    skippedDay: true,
    westernDate: new Date('2013-03-31T12:00:00.000Z'),
    year: {
      rabjungCycle: 17,
      rabjungYear: 27,
      tibYear: 2140,
      westernYear: 2013,
      animal: 'Snake',
      element: 'Water',
      gender: 'Female',
    }
  },
};

const duplicateDayMock = {
  year: 2140,
  month: 2,
  day: 8,
  westernLeap: new Date('2013-03-19T12:00:00.000Z'),
  westernMain: new Date('2013-03-20T12:00:00.000Z'),
  leapDay: {
    day: 8,
    hasLeapDay: true,
    isLeapDay: true,
    julianDate: 2456371,
    month: {
      hasLeapMonth: false, isLeapMonth: false, month: 2, year: 2140
    },
    skippedDay: false,
    westernDate: new Date('2013-03-19T12:00:00.000Z'),
    year: {
      rabjungCycle: 17,
      rabjungYear: 27,
      tibYear: 2140,
      westernYear: 2013,
      animal: 'Snake',
      element: 'Water',
      gender: 'Female',
    }
  },
  mainDay: {
    day: 8,
    hasLeapDay: true,
    isLeapDay: false,
    julianDate: 2456372,
    month: {
      hasLeapMonth: false, isLeapMonth: false, month: 2, year: 2140
    },
    skippedDay: false,
    westernDate: new Date('2013-03-20T12:00:00.000Z'),
    year: {
      rabjungCycle: 17,
      rabjungYear: 27,
      tibYear: 2140,
      westernYear: 2013,
      animal: 'Snake',
      element: 'Water',
      gender: 'Female',
    }
  },
};

const astroMock = {
  day: 11,
  monthCount: 14598,
  meanDate: 2446600.182372702,
  meanSun: 1181.1403399668327,
  moonAnomaly: 1047.3795351473923,
  moonEqu: 16.492063492070884,
  sunAnomaly: 1180.8903399668327,
  sunEqu: -7.263681592026842,
  trueDate: 2446600.578301787,
};

const simpleMonthMock = {
  info: {
    year: 2145,
    month: 1,
    isLeapMonth: false
  },
  fullInfo: {
    year: 2145,
    month: 1,
    isLeapMonth: false,
    hasLeapMonth: false,
    startDate: new Date('2018-02-16T12:00:00.000Z'),
    endDate: new Date('2018-03-17T12:00:00.000Z')
  },
  monthCount: 14990
};

const leapMonthMock = {
  info: {
    year: 2140,
    month: 8,
    isLeapMonth: true
  },
  fullInfo: {
    year: 2140,
    month: 8,
    isLeapMonth: true,
    hasLeapMonth: true,
    startDate: new Date('2013-09-06T12:00:00.000Z'),
    endDate: new Date('2013-10-04T12:00:00.000Z')
  },
  monthCount: 14935
};

const mainMonthMock = {
  info: {
    year: 2140,
    month: 8,
    isLeapMonth: false
  },
  fullInfo: {
    year: 2140,
    month: 8,
    isLeapMonth: false,
    hasLeapMonth: true,
    startDate: new Date('2013-10-05T12:00:00.000Z'),
    endDate: new Date('2013-11-03T12:00:00.000Z')
  },
  monthCount: 14936
};

const yearMock = {
  rabjungCycle: 17,
  rabjungYear: 32,
  westernYear: 2018,
  tibYear: 2145,
  element: 'Earth',
  gender: 'Male',
  animal: 'Dog'
};


export {
  dayInDuplicateMonthMock,
  skippedDayMock,
  duplicateDayMock,
  astroMock,
  simpleMonthMock,
  leapMonthMock,
  mainMonthMock,
  yearMock
};
