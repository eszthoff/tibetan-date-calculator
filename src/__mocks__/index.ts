import {
  simpleMonthMock, leapMonthMock, mainMonthMock
} from './month-mock';
import { dayInDuplicateMonthMock, skippedDayMock, duplicateDayMock } from './day-mock';
import { yearMock } from './year-mock';

const astroMock = {
  day: 11,
  monthCount: 14598,
  meanDate: 2446600.182372702,
  meanSun: 1181.1403399668327,
  moonAnomaly: 1047.3795351473923,
  moonEqu: 16.492063492070884,
  sunAnomaly: 1180.8903399668327,
  sunEqu: -7.263681592026842,
  trueDate: 2446600.578301787,
};

export {
  dayInDuplicateMonthMock,
  skippedDayMock,
  duplicateDayMock,
  astroMock,
  simpleMonthMock,
  leapMonthMock,
  mainMonthMock,
  yearMock
};
