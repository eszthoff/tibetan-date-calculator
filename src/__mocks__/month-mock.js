const simpleMonthMock = {
  info: {
    year: 2145,
    month: 1,
    isLeapMonth: false
  },
  fullInfo: {
    year: 2145,
    month: 1,
    isLeapMonth: false,
    isDoubledMonth: false,
    startDate: '2018-02-16',
    endDate: '2018-03-17'
  },
  monthCount: 14990
};

const leapMonthMock = {
  info: {
    year: 2140,
    month: 8,
    isLeapMonth: true
  },
  fullInfo: {
    year: 2140,
    month: 8,
    isLeapMonth: true,
    isDoubledMonth: true,
    startDate: '2013-09-06',
    endDate: '2013-10-04'
  },
  monthCount: 14935
};

const mainMonthMock = {
  info: {
    year: 2140,
    month: 8,
    isLeapMonth: false
  },
  fullInfo: {
    year: 2140,
    month: 8,
    isLeapMonth: false,
    isDoubledMonth: true,
    startDate: '2013-10-05',
    endDate: '2013-11-03'
  },
  monthCount: 14936
};

export {
  simpleMonthMock,
  leapMonthMock,
  mainMonthMock
};
