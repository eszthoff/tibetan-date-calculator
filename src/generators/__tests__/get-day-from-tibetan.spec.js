import getDayFromTibetan from '../get-day-from-tibetan';
import { dayInDuplicateMonthMock, skippedDayMock, duplicateDayMock } from '../../__mocks__';

describe('getDayFromTibetan()', () => {
  it('should return the correct western date in leap month', () => {
    const {
      year, month, day
    } = dayInDuplicateMonthMock;
    expect(getDayFromTibetan({
      year, month, isLeapMonth: true, day, isLeapDay: false
    })).toMatchSnapshot();
  });
  it('should return the correct western date in non-leap month', () => {
    const {
      year, month, day
    } = dayInDuplicateMonthMock;
    expect(getDayFromTibetan({ year, month, day })).toMatchSnapshot();
  });
  it('should return the correct western date for day marked as leap even though it isn\'t', () => {
    const {
      year, month, day
    } = dayInDuplicateMonthMock;
    expect(getDayFromTibetan({
      year, month, day, isLeapDay: true
    })).toMatchSnapshot();
  });
  it('should return the correct western date for skipped day', () => {
    const {
      year, month, day
    } = skippedDayMock;
    expect(getDayFromTibetan({
      year, month, isLeapMonth: true, day
    })).toMatchSnapshot();
  });
  it('should return the correct western date for leap day', () => {
    const {
      year, month, day
    } = duplicateDayMock;
    expect(getDayFromTibetan({
      year, month, day, isLeapDay: true
    })).toMatchSnapshot();
  });
  it('should return the correct western date for main day that has leap day', () => {
    const {
      year, month, day
    } = duplicateDayMock;
    expect(getDayFromTibetan({ year, month, day })).toMatchSnapshot();
  });
});
