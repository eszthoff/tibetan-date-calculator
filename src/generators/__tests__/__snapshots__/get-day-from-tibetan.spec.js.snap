// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDayFromTibetan() should return the correct western date for day marked as leap even though it isn't 1`] = `
Object {
  "day": 4,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": true,
    "isLeapMonth": false,
    "month": 8,
  },
  "skippedDay": false,
  "westernDate": "2013-10-08",
  "year": 2140,
}
`;

exports[`getDayFromTibetan() should return the correct western date for leap day 1`] = `
Object {
  "day": 8,
  "isDoubledDay": true,
  "isLeapDay": true,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 2,
  },
  "skippedDay": false,
  "westernDate": "2013-03-19",
  "year": 2140,
}
`;

exports[`getDayFromTibetan() should return the correct western date for main day that has leap day 1`] = `
Object {
  "day": 8,
  "isDoubledDay": true,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 2,
  },
  "skippedDay": false,
  "westernDate": "2013-03-20",
  "year": 2140,
}
`;

exports[`getDayFromTibetan() should return the correct western date for skipped day 1`] = `
Object {
  "day": 20,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 2,
  },
  "skippedDay": true,
  "westernDate": "2013-03-31",
  "year": 2140,
}
`;

exports[`getDayFromTibetan() should return the correct western date in leap month 1`] = `
Object {
  "day": 4,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": true,
    "isLeapMonth": true,
    "month": 8,
  },
  "skippedDay": false,
  "westernDate": "2013-09-09",
  "year": 2140,
}
`;

exports[`getDayFromTibetan() should return the correct western date in non-leap month 1`] = `
Object {
  "day": 4,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "month": Object {
    "isDoubledMonth": true,
    "isLeapMonth": false,
    "month": 8,
  },
  "skippedDay": false,
  "westernDate": "2013-10-08",
  "year": 2140,
}
`;
